### PR TITLE
refactor: update text input logic to v2 TextFieldState, part 1 [WPB-8779]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-8779" title="WPB-8779" target="_blank">TaskWPB-8779</a>  Introduce BasicTextField2
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Android Compose has new v2 version of text field inputs, and with that, they are going away from the previous approach with `TextFieldValue` and `onValueChanged` and replacing it with `TextFieldState`.
We already updated text fields to v2 but still are using "hybrid" solution which synchronises between `TextFieldValue/onValueChanged` and `TextFieldState`.

In this PR, following screens, their ViewModels logic and tests are updated to use `TextFieldState`:
- CreateAccountUsernameScreen
- ChangeDisplayNameScreen
- ChangeEmailScreen
- ChangeHandleScreen

Also, `ShakeAnimation` is fixed/updated, `translationX` in `graphicsLayer` is used instead of `offset` to make it lighter and smoother.

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Use text fields from any of listed screens.

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [4764](https://10.10.124.134/job/AR-build-pipeline/4764/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9682055)(9682055)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9682055)(9682055)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/authentication/create/username/CreateAccountUsernameViewModel.kt [hunk 1]
```diff
@@ -30,47 +33,46 @@ import com.wire.kalium.logic.feature.auth.ValidateUserHandleUseCase
...
+                            )
 
-                is ValidateUserHandleResult.Invalid.TooLong -> state.copy(
-                    username = newText.copy(text = textState.handle),
-                    error = HandleUpdateErrorState.None,
-                    continueEnabled = false,
-                    animateUsernameError = false
-                )
+                        is ValidateUserHandleResult.Invalid.InvalidCharacters -> state.copy(
+                                error = HandleUpdateErrorState.None,
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/authentication/create/username/CreateAccountUsernameViewModel.kt:57]: Should it be set to `.None` even if it has invalid characters? 🤔

File: app/src/main/kotlin/com/wire/android/ui/home/settings/account/displayname/ChangeDisplayNameViewModel.kt [hunk 2]
```diff
@@ -18,85 +18,49 @@
...
-                    error = DisplayNameState.NameError.None
-                )
+            getSelf().firstOrNull()?.name.orEmpty().let { currentDisplayName ->
+                textState.setTextAndPlaceCursorAtEnd(currentDisplayName)
+                textState.textAsFlow().collectLatest {
+                    displayNameState = displayNameState.copy(
+                        saveEnabled = it.trim().isNotEmpty() && it.length <= NAME_MAX_COUNT && it.trim() != currentDisplayName,
+                        error = when {
+                            it.trim().isEmpty() -> DisplayNameState.NameError.TextFieldError.NameEmptyError
+                            it.length > NAME_MAX_COUNT -> DisplayNameState.NameError.TextFieldError.NameExceedLimitError
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/settings/account/displayname/ChangeDisplayNameViewModel.kt:57]: Shouldn't we also check if the name is at last the same size or bigger than the minimum characters?

**Review by alexandreferris — COMMENTED** [review]
Amazing work!

got just 2 comments for the returned error type and checking

File: app/src/main/kotlin/com/wire/android/ui/authentication/create/username/CreateAccountUsernameViewModel.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/authentication/create/username/CreateAccountUsernameViewModel.kt:57]: I kept the logic for this one as it was previously, in this particular case we update the text and remove all invalid characters (lines 55-56; we use result from `validateUserHandleUseCase` which returns handle without invalid characters) so `InvalidCharacters` shouldn't matter here and it won't be shown to the user anyway 😄

File: app/src/main/kotlin/com/wire/android/ui/home/settings/account/displayname/ChangeDisplayNameViewModel.kt (hunk 2 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/home/settings/account/displayname/ChangeDisplayNameViewModel.kt:57]: AFAIK display name doesn't have any specific minimum length, it only shouldn't be empty and that's already handled in the line above 🙃

**Review by Garzas — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [4827](https://10.10.124.134/job/AR-build-pipeline/4827/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9731262)(9731262)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9731262)(9731262)-staging-compat.apk)

**Review by ohassine — APPROVED** [review]
LGTM 👌
Just one thing: we are using an alpha version of `compose-foundation` and `compose-material-android`, I'd suggest to upgrade it to latest version `1.7.0-beta01`

**Comment by AndroidBob** [comment]
Build [4888](https://10.10.124.134/job/AR-build-pipeline/4888/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.8.0(9740969)(9740969)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.8.0(9740969)(9740969)-staging-compat.apk)

**Pull request merged**